sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_discovery.py

```diff
@@ -0,0 +1,103 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+import os
+import time
+import dns.resolver
+from dns.resolver import NXDOMAIN, YXDOMAIN, LifetimeTimeout, NoNameservers  # cspell:disable-line
+from dns.rdatatype import SRV  # cspell:disable-line
+from ._constants import DISABLE_APPCONFIGURATION_DISCOVERY, HTTPS_PREFIX
+
+request_retry_period = 5  # seconds
+
+
+class SRVRecord:
+    priority: int
+    weight: int
+    port: int
+    target: str
+
+    def __init__(self, answer):
+        self.priority = answer.priority
+        self.weight = answer.weight
+        self.port = answer.port
+        self.target = str(answer.target).rstrip(".")
+
+
+def find_auto_failover_endpoints(endpoint: str):
+    if os.environ.get(DISABLE_APPCONFIGURATION_DISCOVERY, "").lower() == "true":
+        return []
+    known_domain = _get_known_domains(endpoint)
+    if known_domain is None:
+        return []
+
+    origin = _find_origin(endpoint)
+    if origin is None or not _validate(known_domain, origin.target):
+        return []
+
+    srv_records = [origin] + _find_replicas(origin.target)
+    endpoints = []
+    if endpoint.startswith(HTTPS_PREFIX):
+        endpoint = endpoint[len(HTTPS_PREFIX) :]
```

Comment:
Does python have URL utility methods that can parse the hostname from the URL instead of just intercepting the string?